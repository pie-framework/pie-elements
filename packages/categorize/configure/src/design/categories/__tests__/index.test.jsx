import { shallow } from 'enzyme';
import React from 'react';

import { Categories } from '../index';

describe('Categories', () => {
  let w;
  let onModelChanged = jest.fn();
  let model = {
    choices: [
      {
        id: '0',
        content: 'Choice 0'
      },
    ],
    choicesPosition: 'below',
    choicesLabel: '',
    lockChoiceOrder: true,
    removeTilesAfterPlacing: false,
    categoriesPerRow: 2,
    categories: [
      {
        id: '0',
        label: 'Category 0',
        choices: []
      },
    ],
    rowLabels: [''],
    correctResponse: [],
    partialScoring: true,
  };

  const wrapper = extras => {
    const defaults = {
      classes: {
        categories: 'categories',
        categoriesHolder: 'categoriesHolder',
        row: 'row'
      },
      categories: [{ id: '1', label: 'foo', choices: [] }],
      className: 'className',
      model,
      onModelChanged
    };

    const props = { ...defaults, ...extras };
    return shallow(<Categories {...props} />);
  };

  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();
      expect(w).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    describe('changeCategoryColumns', () => {
      w = wrapper();
      w.instance().changeCategoryColumns({ target: { value: 4 } });

      expect(onModelChanged).toBeCalledWith({ categoriesPerRow: 4 });
    });

    describe('add', () => {
      w = wrapper();
      w.instance().add();

      expect(onModelChanged).toBeCalledWith({
        categories: expect.arrayContaining([{ id: '1', label: 'Category 1' }]),
        rowLabels: ['']
      });
    });

    describe('delete', () => {
      w = wrapper();
      w.instance().delete({ id: '0' });

      expect(onModelChanged).toBeCalledWith(expect.objectContaining({ categories: [] }));
    });

    describe('change', () => {
      it('calls onChange', () => {
        w = wrapper();
        const update = { id: '1', label: 'update', choices: [] };
        w.instance().change(update);
        expect(onModelChanged).toBeCalledWith({ categories: [update] });
      });
    });

    describe('addChoiceToCategory', () => {
      w = wrapper();
      w.instance().addChoiceToCategory({ id: '1', content: 'foo' }, '0');

      expect(onModelChanged).toBeCalledWith({
        correctResponse: [{ category: '0', choices: ['1'] }]
      });
    });

    describe('deleteChoiceFromCategory', () => {
      w = wrapper();
      w.instance().deleteChoiceFromCategory({ id: '0'}, { id: '1' }, 0);

      expect(onModelChanged).toBeCalledWith({
        correctResponse: [{ category: '0', choices: [] }]
      });
    });



  });
});
