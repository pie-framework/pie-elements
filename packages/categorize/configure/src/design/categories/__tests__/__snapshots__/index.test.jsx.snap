// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories snapshot renders 1`] = `
<div
  className="categories className"
>
  <WithStyles(Header)
    buttonLabel="ADD A CATEGORY"
    label="Categories"
    onAdd={[Function]}
  />
  <div
    className="row"
  >
    <TextField
      inputProps={
        Object {
          "max": 4,
          "min": 1,
        }
      }
      label="Categories per row"
      onChange={[Function]}
      required={false}
      select={false}
      type="number"
      value={2}
      variant="standard"
    />
  </div>
  <div
    className="categoriesHolder"
    style={
      Object {
        "gridTemplateColumns": "repeat(2, 1fr)",
      }
    }
  >
    <WithStyles(Component)
      categoriesPerRow={2}
      markup=""
      onChange={[Function]}
      rowIndex={0}
    />
    <WithStyles(Category)
      category={
        Object {
          "choices": Array [],
          "id": "1",
          "label": "foo",
        }
      }
      onAddChoice={[Function]}
      onChange={[Function]}
      onDelete={[Function]}
      onDeleteChoice={[Function]}
    />
  </div>
</div>
`;
